### PR TITLE
appdev: pull required container images

### PR DESCRIPTION
**Pull request by kamaln7**

### CODE REVIEW

File: commands/apps_dev.go [hunk 1]
```diff
@@ -397,3 +413,66 @@ func appDevWorkspace(cmdConfig *CmdConfig) (*workspace.AppDev, error) {
 		AppsService:       cmdConfig.Apps(),
 	})
 }
+
+// PrepareEnvironment pulls required images, validates permissions, etc. in preparation for a component build.
+func appDevPrepareEnvironment(ctx context.Context, ws *workspace.AppDev, cli builder.DockerEngineClient, componentSpec godo.AppBuildableComponentSpec) error {
+	cnbBuilderImage := builder.CNBBuilderImage
```
Comment (nicktate) [inline commands/apps_dev.go:419]: We need to pre-pull the `apps-run` image; not sure the best way to infer this info based on the `CNBBuilderImage` ; perhaps hard code temporarily but we should be able to fetch it from the image metadata of the builder after we pull it.

File: internal/apps/builder/builder.go [hunk 2]
```diff
@@ -19,6 +19,9 @@ import (
 	"github.com/docker/docker/pkg/stdcopy"
 )
 
+// StaticSiteNginxImage is the nginx image used for static site container images.
+const StaticSiteNginxImage = "nginx:alpine"
```
Comment (nicktate) [inline internal/apps/builder/builder.go:23]: should we pin this to a specific alpine version (or at least a major version line)?

**Review by nicktate — COMMENTED** [review]
This looks great overall; main comment is around the need to pre-pull `digitaloceanapps/apps-run:...` as well

**Review by ChiefMateStarbuck — COMMENTED** [review]
Good on Green

**Review by nicktate — APPROVED** [review]

**Review by ChiefMateStarbuck — APPROVED** [review]

**Pull request merged**